src/main/java/com/google/api/codegen/CodeGeneratorApi.java

```diff
@@ -61,9 +59,14 @@
           "The list of YAML configuration files for the code generator.",
           ImmutableList.<String>of());
 
-  /**
-   * Constructs a code generator api based on given options.
-   */
+  public static final Option<Boolean> DISABLE_TEST_GEN =
```

Comment:
I much prefer a positively-worded term. Maybe instead we could do "filter_artifacts", and support the values "surface" and "test"; if nothing is given, generate everything; for users who only want the surface, they pass in "--filter_artifacts=surface".